File: apportable/src/com/anotherplace/java/KamcordBridge.java
package com.anotherplace.java;

import com.anotherplace.java.*;

import android.util.Log;
import android.content.Context;
import android.app.Activity;
import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Intent;
import android.app.AlertDialog;
import android.app.AlertDialog.Builder;
//import android.R;

import com.kamcord.android.Kamcord;

public class KamcordBridge
{
	Activity mainActivity;
    
    private static final String TAG = "KamcordBridge";

    public KamcordBridge(Activity activity, String key, String secret)
    {
		mainActivity = activity;
        
        Kamcord.initActivity(mainActivity);
        Kamcord.initKeyAndSecret(key, secret, "Dragon Finga");        
    }
    
    public void beginDraw()
    {
        Kamcord.beginDraw();
    }
    
    public void endDraw()
    {
        Kamcord.endDraw();
    }
    
    public void startRecording()
    {
        Log.d(TAG, "startRecording");
        Kamcord.startRecording();
    }
    
    public void stopRecording()
    {
        Log.d(TAG, "stopRecording");
        Kamcord.stopRecording();
    }
    
    public void pauseRecording()
    {
        Log.d(TAG, "pauseRecording");
        Kamcord.pauseRecording();
    }
    
    public void resumeRecording()
    {
        Log.d(TAG, "resumeRecording");
        Kamcord.resumeRecording();
    }
    
    public void showView()
    {
        Log.d(TAG, "showView");
        Kamcord.showView();
    }
	
	public boolean isEnabled()
	{
		return Kamcord.isEnabled();
	}
}

File: Test/KamcordBridge.m
#include "KamcordBridge.h"
#import "AppDelegate.h"

#ifdef ANDROID

@implementation Kamcord

+(void)beginDraw
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge beginDraw];
}

+(void)endDraw
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge endDraw];
}

+(void)startRecording
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge startRecording];
}

+(void)stopRecording
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge stopRecording];
}

+(void)pause
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge pauseRecording];
}

+(void)resume
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge resumeRecording];
}

+(void)showView
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge showView];
}

+(void)isEnabled
{
    AppDelegate* appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
    [appDelegate.kamcordBridge isEnabled];
}

@end

////////////////////////////////////////////////////////////////////////////////////////////////////
// KamcordBridge
////////////////////////////////////////////////////////////////////////////////////////////////////

@implementation KamcordBridge

//--------------------------------------------------------------------------------------------------
+ (void)initializeJava
{
    // Note: this must be called for any class that registers custom
    // java apis, without this call the inheritance may not work as expected
    [super initializeJava];
	
    // Bridge registration methods must be called on the class and NOT self
    // even though that this is a static method (this preserves inheritance
    // to the correct java class
    [KamcordBridge registerConstructorWithSelector:@selector(initWithActivity:Key:Secret:)
                                      arguments:[AndroidActivity className], [NSString className], [NSString className], nil];
    
    [KamcordBridge registerInstanceMethod:@"beginDraw" selector:@selector(beginDraw) arguments:nil];
    [KamcordBridge registerInstanceMethod:@"endDraw" selector:@selector(endDraw) arguments:nil];

    [KamcordBridge registerInstanceMethod:@"startRecording" selector:@selector(startRecording) arguments:nil];
    [KamcordBridge registerInstanceMethod:@"stopRecording" selector:@selector(stopRecording) arguments:nil];
    [KamcordBridge registerInstanceMethod:@"pauseRecording" selector:@selector(pauseRecording) arguments:nil];
    [KamcordBridge registerInstanceMethod:@"resumeRecording" selector:@selector(resumeRecording) arguments:nil];

    [KamcordBridge registerInstanceMethod:@"isEnabled" selector:@selector(isEnabled) returnValue:[JavaClass boolPrimitive]];

    [KamcordBridge registerInstanceMethod:@"showView" selector:@selector(showView) arguments:nil];
}

//--------------------------------------------------------------------------------------------------
+ (NSString *)className
{
    return @"com.anotherplace.java.KamcordBridge";
}

@end

#endif